packages/shared-ux/chrome/navigation/src/ui/v2/types.ts

```diff
@@ -6,27 +6,172 @@
  * Side Public License, v 1.
  */
 
-import { ChromeNavLink, ChromeProjectNavigationNode } from '@kbn/core-chrome-browser';
-import { ReactNode } from 'react';
+import type { ReactElement, ReactNode } from 'react';
+import type {
+  ChromeProjectNavigationLink,
+  ChromeProjectNavigationNode,
+} from '@kbn/core-chrome-browser';
 
-export interface InternalNavigationNode extends Omit<ChromeProjectNavigationNode, 'id' | 'link'> {
-  id: string;
-  title: string;
-  deepLink?: ChromeNavLink;
+import type { CloudLinkProps, RecentlyAccessedProps } from './components';
+
+/**
+ * @public
+ *
+ * A navigation node definition with its unique id, title, path in the tree and optional
+ * deep link and children.
+ */
+export interface NodeDefinition<T extends string = string, C extends string = T> {
+  /** Optional id, if not passed a "link" must be provided. */
+  id?: T;
+  /** Optional title. If not provided and a "link" is provided the title will be the Deep link title */
+  title?: string;
+  /** App id or deeplink id */
+  link?: ChromeProjectNavigationLink;
+  /** Optional icon for the navigation node. Note: not all navigation depth will render the icon */
+  icon?: string;
+  /** Optional children of the navigation node */
+  children?: Array<NodeDefinition<C>>;
+}
+
+/**
+ * @public
+ *
+ * A navigation node definition with its unique id, title, path in the tree and optional.
+ * Those are the props that can be passed to the Navigation.Group and Navigation.Item components.
+ */
+export interface NodeProps extends Omit<NodeDefinition, 'children'> {
+  /**
+   * Children of the node. For Navigation.Item (only) it allows a function to be set.
+   * This function will receive the
+   */
+  children?: ((navNode: ChromeProjectNavigationNode) => ReactNode) | ReactNode;
+}
+
+/**
+ * @private
+ *
+ * Internally we enhance the Props passed to the Navigation.Item component.
+ */
+export interface NodePropsEnhanced extends NodeProps {
+  /**
+   * This function correspond to the same "itemRender" function that can be passed to
+   * the EuiSideNavItemType (see navigation_section_ui.tsx)
```

Comment:
In EuiSideNavItemType, the prop is called `renderItem` not `itemRender`. Can we give ours the same name?

Reply:
Good catch.. I really meant to use the same naming! 😄 

Addressed in https://github.com/elastic/kibana/pull/158297/commits/8a258d70f4733d61abc5405b9570594e85e69381